jupyterhub_singleuser_profiles/sizes.py

```diff
@@ -30,6 +30,14 @@ def get_form(self, last_size=None):
             selected = ""
             if size['name'] == last_size:
                 selected = "selected=selected"
-            options.append("<option value='%s' %s>%s</option>" % (size['name'], selected, "%s (Memory: %s, CPU: %s)" % (size['name'], size['resources']['mem_limit'], size['resources']['cpu_limit'])))
+            if size['resources'].get('mem_limit') or size['resources'].get('cpu_limit'):
+                # Kept for backwards compatibility
+                options.append("<option value='%s' %s>%s</option>" % (size['name'], selected, "%s (Memory: %s, CPU: %s)" % (size['name'], size['resources']['mem_limit'], size['resources']['cpu_limit'])))
+            else:
+                req_mem = size['resources']['requests']['memory']
```

Comment:
This and/or next 3 lines will fail if user specifies only `request` or only `limits` - which is fine for Kubernetes.

I would suggest to implement some logic which will check if there is only one of `requests` or `limits` and sets the other to the equal values